### ISSUE TITLE
Question about AUC values on test datasets

### ISSUE THREAD

**Issue by muchatang** [rainy-xu/TALL4Deepfake#6]
Thanks for your wonderful work. 
I trained the TALL_Swin model myself on ff++ dataset using the training code you provided and tested it with test.py file.
The train loss value at the 60th epoch is 0.2053, and the AUC value on the ff++ validation set is 0.99430683049763. 
However, the test results are quite different from your results in the paper, 98.30 on the ff++ dataset and 82.27 on the celeb-DF dataset. 
What could be the reason for the big difference in the results? 
Meanwhile，in the main.py file, the value of '--frames_per_group' is set to 3. 
However, in the paper, consecutive frames are sampled to form one clip. Should the value of '--frames_per_group' be altered to 1？

**Comment by rainy-xu** [comment]
> Thanks for your wonderful work. I trained the TALL_Swin model myself on ff++ dataset using the training code you provided and tested it with test.py file. The train loss value at the 60th epoch is 0.2053, and the AUC value on the ff++ validation set is 0.99430683049763. However, the test results are quite different from your results in the paper, 98.30 on the ff++ dataset and 82.27 on the celeb-DF dataset. What could be the reason for the big difference in the results? Meanwhile，in the main.py file, the value of '--frames_per_group' is set to 3. However, in the paper, consecutive frames are sampled to form one clip. Should the value of '--frames_per_group' be altered to 1？

Hi, perhaps you could try more epochs. 
Additionally, differences in data preprocessing methods can cause variances in results,  you may need to remove frames that were preprocessed incorrectly.
Thanks for the reminder, “--frames_per_group” should be set to 1.

**Comment by YangYangGirl** [comment]
Thanks for the provided clean code and your excellent work. I set the “--frames_per_group” to 1, removed frames that were preprocessed incorrectly, and renamed the index of the frames making them consecutive as you said. However, I found the test results are lower than your results in the paper, 97.06 on the ff++ dataset and 81.47 on the celeb-DF dataset. I provided the train scripts below:
python main_hdf5.py --data_dir data --data_txt_dir dataset_txt --dataset ffpp \
	 --input-size 112 --num_clips 8 --output_dir output/train_hdf5 --opt adamw --lr 1.5e-5 --warmup-lr 1.5e-8 --min-lr 1.5e-7 \
	  --epochs 60 --sched cosine --duration 4 --batch-size 4 --thumbnail_rows 2 --disable_scaleup --cutout True \
	   --pretrained --warmup-epochs 10 --no-amp --model TALL_SWIN \
	    --hpe_to_token > ./output/train_hdf5/train_hdf5_ffpp.log

**Comment by YangYangGirl** [comment]
I also tried another strategy where I sampled 8 clips from **the longest video segment** where a face could be detected, the test results would be higher, 97.52 on the ff++ dataset and 84.03 on the celeb-DF dataset. But the results are still not as high as your paper. I don't know if there's something else I'm overlooking.

**Comment by muchatang** [comment]
I also have a similar problem that hasn't been solved yet. But this issue is now closed. Maybe you can start a new one to ask the author.
